tests/component/test_task_events.py

```diff
@@ -240,21 +296,69 @@ def test___every_n_samples_transferred_from_buffer_event_registered___register_e
     )
 
 
-@pytest.mark.library_only
+@pytest.mark.xfail(
+    reason="AB#2395958: GrpcStubInterpreter does not report event registration errors to caller"
+)
 def test___signal_event_registered___register_signal_event___already_registered_error_raised(
     ai_task: nidaqmx.Task,
 ) -> None:
     event_observer = SignalEventObserver()
     ai_task.register_signal_event(Signal.SAMPLE_COMPLETE, event_observer.handle_signal_event)
-    ai_task.timing.cfg_samp_clk_timing(rate=10000.0, samps_per_chan=1000)
+    ai_task.timing.cfg_samp_clk_timing(
+        rate=10000.0, sample_mode=AcquisitionType.FINITE, samps_per_chan=1000
+    )
 
     with pytest.raises(nidaqmx.DaqError) as exc_info:
         ai_task.register_signal_event(Signal.SAMPLE_COMPLETE, event_observer.handle_signal_event)
 
     assert exc_info.value.error_code == DAQmxErrors.SIGNAL_EVENT_ALREADY_REGISTERED
 
 
-@pytest.mark.library_only
+@pytest.mark.xfail(
```

Comment:
Doesn't this test pass for lib? So would show up as "unexpectedly passing"?

Reply:
Yes, for library it will have result XPASS. A grpc_xfail(reason) mark would be appropriate for most of these.